runtime/cudaq/platform/default/rest/Executor.cpp

```diff
@@ -35,18 +35,15 @@ Executor::execute(std::vector<KernelExecution> &codesToExecute) {
                 response.dump());
 
     // Add the job id and the job name.
-
     auto task_id = serverHelper->extractJobId(response);
-    config["output_names." + task_id] = codesToExecute[i].output_names.dump();
-
     if (task_id.empty()) {
       nlohmann::json tmp(job.at("tasks"));
       serverHelper->constructGetJobPath(tmp[0]);
-      ids.emplace_back(
-          std::make_pair(tmp[0].at("task_id"), codesToExecute[i].name));
-    } else {
-      ids.emplace_back(task_id, codesToExecute[i].name);
+      task_id = tmp[0].at("task_id");
     }
+    cudaq::info("Task ID is {}", task_id);
+    ids.emplace_back(task_id, codesToExecute[i].name);
+    config["output_names." + task_id] = codesToExecute[i].output_names.dump();
```

Comment:
`dump()` isn't a good name for a function returning a `string`. `str()` would be better. I don't know if we own the code this references.

Reply:
That object is of type `nlohmann::json`, so we do not own that code.